sqlmesh/core/dialect.py

```diff
@@ -759,3 +763,17 @@ def find_tables(expression: exp.Expression, dialect: DialectType = None) -> t.Se
         for table in scope.tables
         if not isinstance(table.this, exp.Func) and exp.table_name(table) not in scope.cte_sources
     }
+
+
+def transform_values(
+    values: t.Tuple[t.Any, ...], columns_to_types: t.Dict[str, exp.DataType]
+) -> t.Iterator[t.Any]:
+    """Perform transformations on values given columns_to_types.
+
+    Currently, the only transformation is wrapping JSON columns with PARSE_JSON().
+    """
+    for value, col_type in zip(values, columns_to_types.values()):
+        if col_type == exp.DataType.build("json"):
+            yield exp.func("PARSE_JSON", f"'{value}'")
+        else:
+            yield value
```

Comment:
Should we traverse `columns_to_types` once so that we do this transformation using indexes instead of iterating over the whole tuple each time? I'm a bit skeptical about the performance of this.

Reply:
Won't it be the same number of iterations since we have to iterate over every element in the `values` tuple anyway?

Reply:
Hmm maybe, unless we mutate `values`?

Reply:
I don't think we can since `values` is a tuple.

Reply:
Ah, that's right, I was under the impression we used a list. Please ignore me then.